src/applications/lgy/coe/form/containers/ConfirmationPage.jsx

```diff
@@ -20,8 +20,11 @@ const ConfirmationPage = ({ form }) => {
   });
 
   const { data, submission } = form;
-  const name = data.fullName;
-  const { referenceNumber } = submission.response.attributes || '';
+  const name = data?.fullName || {};
+  const fullName = `${name.first} ${name.middle || ''} ${name.last}`;
+
+  const submitDate = moment(submission?.timestamp || Date.now());
+  const { referenceNumber } = submission?.response?.attributes || {};
```

Comment:
it should be a string by default, no?

`const { referenceNumber } = submission?.response?.attributes || '';`

Reply:
you mean like this?

```js
const { referenceNumber = '' } = submission?.response?.attributes || {};
```

the empty object is the fallback for an undefined submission response